DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1

```diff
@@ -312,52 +315,69 @@ function EnsureVHDState
 # Change the Cim Instance objects in to a hash table containing property value pair.
 function GetItemToCopy
 {
-    param([Microsoft.Management.Infrastructure.CimInstance] $item)
```

Comment:
Got sick of looking at this repetitive code as I was staring at it, fixing the validation errors. I decided it was better to rewrite it with loops instead of just swapping the $null references to the left side of the comparison.